components/widgets/resources/src/main/resources/resources/uicomponents/widgets/extraGeneVariantData.js

```diff
@@ -427,6 +439,14 @@ var ExtraGeneVariantData = (function (ExtraGeneVariantData) {
       });
     },
 
+    restrictNumericInput : function () {
+      this._geneTable.select('input:[id$=_start_position]', 'input:[id$=_end_position]').each( function(input) {
+        input.observe('keyup', function(event){
+          this.value=this.value.replace(/[^\d]/,'');
```

Comment:
Duplicated code + bug: this copy of the code which restricts inputs to numerical values doesn't have the global flag and will have the issues I reported in the previous review.

Must change: `this.value = this.value.replace(/[^\d]/g,'');`.

Suggestion for further improvement:

```
restrictNumericInput : function (element) {
  (element || this._geneTable).select('input:[id$=_start_position]', 'input:[id$=_end_position]').invoke('observe', 'keyup', function(event) {
    this.value = this.value.replace(/[^\d]/g,'');
  });
}
```

Call `this.restrictNumericInput()` (no parameter, defaults to `this._geneTable`) in the initialization, and `this.restrictNumericInput(newMoreInfoRow)` when adding a new variant.